src/core/mac/mac.hpp

```diff
@@ -769,10 +802,13 @@ class Mac : public InstanceLocator
     ChannelMask   mSupportedChannelMask;
     ExtendedPanId mExtendedPanId;
     NetworkName   mNetworkName;
-    uint8_t       mScanChannel;
-    uint16_t      mScanDuration;
-    ChannelMask   mScanChannelMask;
-    uint8_t       mMaxFrameRetriesDirect;
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+    DomainName mDomainName;
```

Comment:
Is it necessary to reorder data members of `class Mac` so that they are better aligned on boundaries according to their types?

Reply:
Nice to have.  Let's keep this PR simple as is for now.
Would you please help to submit a separate PR for reorder?

Reply:
Neither do I think we should handle it in this PR.